src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,162 @@
+        userWithNullPassword.setPassword(null);
+        userWithNullPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullPassword);
+        });
+    }
+
+    @Test
+    void register_emptyPassword_notOk() {
+        User userWithEmptyPassword = new User();
+        userWithEmptyPassword.setLogin("validLogin");
+        userWithEmptyPassword.setPassword("");
+        userWithEmptyPassword.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyPassword);
+        });
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User userWithShortLogin = new User();
+        userWithShortLogin.setLogin("log");
+        userWithShortLogin.setPassword("validPassword");
+        userWithShortLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithShortLogin);
+        });
+    }
+
+    @Test
+    void register_minLengthLogin_ok() {
+        User userWithMinLengthLogin = new User();
+
+        String minLengthLogin = "MinLog";
+        userWithMinLengthLogin.setLogin(minLengthLogin);
+        userWithMinLengthLogin.setPassword("validPassword");
+        userWithMinLengthLogin.setAge(20);
+
+        registrationService.register(userWithMinLengthLogin);
+        assertEquals(userWithMinLengthLogin, storageDao.get(minLengthLogin));
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        User userWithEmptyLogin = new User();
+        userWithEmptyLogin.setLogin("");
+        userWithEmptyLogin.setPassword("validPassword");
+        userWithEmptyLogin.setAge(20);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithEmptyLogin);
+        });
+    }
+
+    @Test
+    void register_ageIsZero_notOk() {
+        User userWithNullAge = new User();
+        userWithNullAge.setLogin("validLogin");
+        userWithNullAge.setPassword("validPassword");
+        userWithNullAge.setAge(0);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithNullAge);
+        });
+    }
+
+    @Test
+    void register_fewYears_notOk() {
+        User userWithFewYears = new User();
+        userWithFewYears.setLogin("validLogin");
+        userWithFewYears.setPassword("validPassword");
+        userWithFewYears.setAge(16);
+
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(userWithFewYears);
+        });
+    }
+
+    @Test
+    void register_minAge_ok() {
+        User userWithMinAge = new User();
+        userWithMinAge.setLogin("LoginV");
+        userWithMinAge.setPassword("PasswordV");
+        userWithMinAge.setAge(18);
+        registrationService.register(userWithMinAge);
+        assertEquals(userWithMinAge, storageDao.get("LoginV"));
```

Comment:
In the `register_minAge_ok` test case, you are not testing the edge case. According to the checklist, it would be good to test for an age of 18 (which is the minimum valid age).